modyn/selector/internal/selector_strategies/abstract_selection_strategy.py

```diff
@@ -47,6 +48,12 @@ def __init__(
         self.has_limit = self.training_set_size_limit > 0
         self.reset_after_trigger: bool = config["reset_after_trigger"]
 
+        # weighted optimization is quite unusual, so the default us false
+        if "uses_weights" in config:
+            self._uses_weights = config["uses_weights"]
```

Comment:
In my opinion, it can be a pipeline parameter. For example, if we have a dataset of autonomous driving and we want to see if the weights give an advantage. We can compare the results of uses_weights = True with those of uses_weights = False regardless of the strategy. Then, we can raise a ValueError in the specific strategy if uses_weights is True but it cannot be (like in remote downsamplers).